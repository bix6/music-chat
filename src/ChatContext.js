import React from 'react';

const ChatContext = React.createContext({
    name: '',
    updateName: () => {},
    chatroomList: [],
    currentChatroom: '',
    updateChatroom: () => {},
    messages: {}
});

export default ChatContext;