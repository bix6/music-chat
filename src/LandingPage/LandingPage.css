/* App Styles */
.landing-page-main {
  text-align: center;
  padding: 0 1rem;
  margin: 1rem auto;
  max-width: 700px;
  border: 3px solid #43cbff;
  background-image: linear-gradient(20deg, #9708cc 10%, #43cbff 100%);
}

.landing-page-main p {
  font-size: 2rem;
  line-height: 4rem;
  color: black;
}

.landing-page-main form {
  margin-top: 1.5rem;
}

.landing-page-main label {
  display: block;
  margin: 0rem auto;
  font-size: 2rem;
  color: #44ffd2;
}

.landing-page-main input {
  padding: 0.3rem 0 0.3rem 1rem;
  margin: 0.5rem auto;
}

.landing-page-main button {
  display: block;
  margin: 0.5rem auto;
}
