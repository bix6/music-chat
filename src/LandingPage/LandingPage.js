import React from 'react';
import './LandingPage.css';
import ValidationError from '../shared-components/ValidationError/ValidationError';
import ChatContext from '../ChatContext';
import HeaderComponent from '../HeaderComponent/HeaderComponent';

class LandingPage extends React.Component {
    static contextType = ChatContext;

    state = {
        touched: false
    };

    updateName = (name) => {
        this.setState({
            touched: true
        });

        this.context.updateName(name)
    }

    validateName = () => {
        const name = this.context.name.trim();

        if (name.length < 2) {
            return 'Name must be 2 characters or more'
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        this.props.history.push('/chat');
    }

    render() {
        return (
            <>
                <HeaderComponent />
                <main>
                    <p>Share music across the globe. <br />Enter a name to get started.</p>
                    <form onSubmit={ e => this.handleSubmit(e) }>
                        <label htmlFor="choose-name">Name / Nickname:</label>
                        <input type="text" id="choose-name" name="choose-name" 
                            onChange= { e =>  this.updateName(e.target.value) }/>
                        <button type="submit" disabled={ this.validateName() }>Chat</button>
                        { this.state.touched && <ValidationError message={ this.validateName() }/> }
                    </form>
                </main>
            </>
        )
    }
}

export default LandingPage;