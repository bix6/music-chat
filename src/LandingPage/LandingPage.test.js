import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import LandingPage from './LandingPage';
import ChatContext from '../ChatContext';

describe('LandingPage Component', () => {
    it('Smoke Test: Renders Empty', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <LandingPage />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Snapshot Test: Default, Empty Name', () => {
        const TestComponent = () => (
            <ChatContext.Provider value={{ name: '' }}>
                <LandingPage />
            </ChatContext.Provider>
        )
        const wrapper = mount(<TestComponent />);
        expect(toJson(wrapper.find(LandingPage))).toMatchSnapshot();
        wrapper.unmount();
    })
})