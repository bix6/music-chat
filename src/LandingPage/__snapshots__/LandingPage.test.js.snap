// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component Snapshot Test: Default, Empty Name 1`] = `
<LandingPage>
  <HeaderComponent>
    <header>
      <h1>
        Music Chat
      </h1>
    </header>
  </HeaderComponent>
  <main>
    <p>
      Share music across the globe. Choose a name to get started.
    </p>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="choose-name"
      >
        Name / Nickname:
      </label>
      <input
        id="choose-name"
        name="choose-name"
        onChange={[Function]}
        type="text"
      />
      <button
        disabled="Name must be 2 characters or more"
        type="submit"
      >
        Chat
      </button>
    </form>
  </main>
</LandingPage>
`;
